src/e2e/puppeteer/helpers/dragAndDropThought.ts

```diff
@@ -0,0 +1,62 @@
+import { Page } from 'puppeteer'
+import sleep from '../../../util/sleep'
+import getEditable from './getEditable'
+
+interface DragAndDropOptions {
+  /** Determines where the destination thought is dropped, relative to the source thought.
+    - after: drop the source thought as a sibling after the destination thought.
+    - before: drop the source thought as a sibling before the destination thought.
+    - child: drop the source thought as a child of the destination thought.
+   */
+  position: 'after' | 'before' | 'child'
+}
+
+/** Performs Drag and Drop functionality on a thought in Puppeteer browser. */
+const dragAndDropThought = async (
+  page: Page,
+  sourceValue: string,
+  destValue: string,
+  { position }: DragAndDropOptions,
+) => {
+  if (position === 'before') {
+    throw new Error('Not implemented')
+  } else if (position === 'child') {
+    throw new Error('Not implemented')
+  }
+
+  const sourceElement = await getEditable(page, sourceValue)
+  const destElement = await getEditable(page, destValue)
+
+  const dragStart = await sourceElement.boundingBox()
+  const dragEnd = await destElement.boundingBox()
+
+  if (!dragStart) {
+    throw new Error('Drag source element not found')
+  } else if (!dragEnd) {
+    throw new Error('Drag destination element not found')
+  }
+
+  // Offset to add to drop destination y position so it is dragged exactly on top of the drop traget.
+  // Otherwise it falls short, just above the drop target.
+  const yOffset = 20
+
+  // Calculate center positions of the elements
+  const dragPosition = {
+    x: dragStart.x + 1,
+    y: dragStart.y + 1,
+  }
```

Comment:
I had to add 1 to the x and y positions to get the drag to trigger on MacOS.

Reply:
Sure thing, updating snapshots and pushing them. Hmm, not sure why is this maybe the rendering difference between the platforms.

Reply:
All checks have passed, please merge when you get the chance.